### PR TITLE
Add head sampling section + elaborate tail sampling

### PR DESCRIPTION
**Pull request by cartermp**
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/2457

This still leaves our sampling docs in a very, very incomplete state. But in the name of incremental progress, here's something.

Preview: https://deploy-preview-2463--opentelemetry.netlify.app/docs/concepts/sampling/

### CODE REVIEW

File: content/en/docs/concepts/sampling/index.md [hunk 1]
```diff
@@ -30,18 +30,93 @@ want to sample. You might want to customize your sampling strategy to:
...
+
+As you can see, Tail Sampling allows for a much higher degree of sophistication.
+For larger systems that must sample telemetry, it is almost always necessary to
+use Tail Sampling to balance data volume with usefulness of that data.
+
+There are three primary downsides to Tail Sampling today:
+
+- Tail Sampling can be difficult to implement. Depending on the kind of sampling
+  techniques available to you, it is not always a "set and forget" kind of
+  setting. As your systems change, so too will your sampling strategies. For a
```
Comment (kentquirk) [inline content/en/docs/concepts/sampling/index.md:100]:
```suggestion
  thing. As your systems change, so too will your sampling strategies. For a
```

File: content/en/docs/concepts/sampling/index.md [hunk 2]
```diff
@@ -30,18 +30,93 @@ want to sample. You might want to customize your sampling strategy to:
...
+  large and sophisticated distributed system, rules that implement sampling
+  strategies can also be large and sophisticated
```
Comment (kentquirk) [inline content/en/docs/concepts/sampling/index.md:102]:
```suggestion
  strategies can also be large and sophisticated.
```

**Review by chalin — COMMENTED** [review]
I'd avoid capitalizing "head sampling" and "tail sampling" (unless the context asks for it like a heading or the start of a sentence).

File: content/en/docs/concepts/sampling/index.md [hunk 3]
```diff
@@ -30,18 +30,93 @@ want to sample. You might want to customize your sampling strategy to:
...
+state that they are "sampling out data" or that data not processed or exported
+is considered "sampled". These are incorrect statements.
+
+## Head Sampling
+
+Head sampling is a sampling technique used to make a sampling decision as early
+as possible. A decision to sample or drop a span or trace is not made by
+inspecting the trace as a whole.
+
+For example, the most common form of head sampling is Consistent Probability
```
Comment (svrnm) [inline content/en/docs/concepts/sampling/index.md:54]: can we link to the spec of Consistent Probability Sampling here?

**Review by svrnm — APPROVED** [review]
non-blocking comment, great addition:)

File: content/en/docs/concepts/sampling/index.md (hunk 3 above)
Comment (cartermp) [reply content/en/docs/concepts/sampling/index.md]: Doneski!

**Comment by cartermp** [comment]
Hmmm, after I hit the button I wondered if the head sampling bit should not use the Consistent Probability sampler terminology, since technically SDKs don't implement this yet.

...but, like, the concept is still the same from the perspective of an end-user. Deterministically taking a fraction of traces is what both do, but the CPS propagates metadata that tells you information about the total population that the sample comes from. This makes the CPS superior because it lets a telemetry backend re-weight counts so that any aggregation you derive from tracing data (like total # of requests) is actually correct w.r.t. the total population compared to the sample.

...anyways, I'm really not too worried at the moment, as it's easy enough to change. Frankly this is the direction all the head samplers should be heading anyways.

**Review by jmacd — COMMENTED** [review]
LGTM

**Pull request merged**